### DISCUSSION TITLE
[Case Study] How applying SCI to your system can affect the reliability?

### DISCUSSION THREAD

**Discussion by greenhsu123** [Green-Software-Foundation/sci#211] [Ideas]
[Case Study] How applying SCI to your system can affect the reliability?

**Comment by dubrie** [comment]
A few thoughts on how SCI can impact a systems reliability:
1. Reliability has an indirect relationship to a systems complexity so the less complex the system is, the more reliability it typically has. Every piece of a system has a failure rate and the more pieces in the system the high chance something will fail and affect the overall reliability. 
2. One of the ways to make a system less complex is to "right-size" utilization of the resources to reduce to total number of machines, nodes, databases, etc. I use "right-size" instead of maximize because redundancy and surge capacity may be required in some systems in order to provide high reliability
3. Reliability is commonly measured in error rates in the system. A high error rate can trigger retry requests which drives up the overall traffic and overall resource usage. A focus on low error rates will reduce the amount of network traffic, machines, log file sizes, and many other aspects of the systems that are responsible for emissions.
4. Another common reliability metric is latency in the system. Low latency reduces bottlenecks and wait times on clients and servers which reduces the total number of resources required to server requests to APIs, machines, and databases. 
5. Likely outside this use case, but running CPUs at around 70% of their max clockspeed leads to longer lifetimes and reduced embodied carbon overall.

Reliability purely through redundancy is one approach for a system but will result in high costs and high carbon intensity. I believe that this use case can show that driving for reliability through the SCI and intentionally focusing on your systems carbon intensity will produce a less complex system, reduced resource requirements, higher resiliency, and even higher performance overall.
